lib/components/molecules/DateSelector/DateSelector.js

```diff
@@ -0,0 +1,184 @@
+// @flow
+/**
+ *
+ * DateSelector
+ *
+ */
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+import classNames from 'classnames';
+import styles from './DateSelector.style';
+import type { Props } from './types';
+import Select from '../../atoms/Select';
+import Label from '../../atoms/Label';
+
+const DateSelector = ({
+  id,
+  className,
+  format,
+  startDate,
+  endDate,
+  locale,
+  dateLabel,
+  monthLabel,
+  yearLabel,
+}: Props): Node => {
+  const [selectedDate, setSelectedDate] = useState(new Date());
+  const [months, setMonths] = useState([]);
+
+  const getMonths = () => {
+    const date = new Date();
+    const monthsOptions = [];
+    for (let i = 0; i < 12; i += 1) {
+      date.setMonth(i);
+      monthsOptions.push(date.toLocaleString(locale, { month: 'long' }));
+    }
+    return monthsOptions;
+  };
+
+  useEffect(() => {
+    setMonths(getMonths());
+  }, []);
+
+  useEffect(() => {
+    setSelectedDate(startDate);
+  }, [startDate, endDate]);
+
+  const getDaysInMonth = () => {
+    const daysOptions = [];
+    let startDay = 1;
+    let endDay = new Date(selectedDate.getFullYear(), selectedDate.getMonth() + 1, 0).getDate();
+    if (
+      selectedDate.getMonth() === startDate.getMonth() &&
+      selectedDate.getFullYear() === startDate.getFullYear()
+    ) {
+      startDay = startDate.getDate();
+    }
+    if (
```

Comment:
Still there is redundant code we can reduce, for example:

1.  These 2 If conditions were we are startDay and endDay could be returned by a function.